network/server.go

```diff
@@ -665,33 +666,25 @@ func (s *Server) SubscribeFn(handler func(evnt *PeerEvent)) error {
 func (s *Server) SubscribeCh() (<-chan *PeerEvent, error) {
 	ch := make(chan *PeerEvent)
 
-	var closed bool
-
-	var mutex sync.Mutex
-
-	isClosed := func() bool {
-		mutex.Lock()
-		defer mutex.Unlock()
-
-		return closed
-	}
+	var isClosed int32 = 0
 
 	err := s.SubscribeFn(func(evnt *PeerEvent) {
-		if !isClosed() {
+		fmt.Println("usaopoo")
+		if atomic.LoadInt32(&isClosed) == 0 {
+			fmt.Println("prosaoo")
 			ch <- evnt
 		}
 	})
 	if err != nil {
+		atomic.AddInt32(&isClosed, 1)
```

Comment:
No need to use `AddInt32`, you can use `StoreInt32`

Reply:
Fixed 9c647477d495aa6642852e296dd7c753d44c293c